lumicks/pylake/channel.py

```diff
@@ -504,6 +504,172 @@ def plot(self, start=None, **kwargs):
+
+            # Draws an annotation at the 1-second mark
+            slc.highlight_time_range("1s", annotation="first second")
+
+            # Draws a highlighted region from 1 to 3 seconds and marks it with the text "region"
+            slc.highlight_time_range(("1s", "3s"), annotation="region")
+
+            # Draws a highlighted region from 4 to 5 seconds, marks it, but forces text vertically
+            slc.highlight_time_range(("1s", "3s"), annotation="region", annotation_direction="vertical")
+
+
+            plt.figure()
+            f.force1x.plot()
+
+            item = f.force1x.calibration[1]
+
+            # Fills the region between calibration_item.start and calibration_item.stop
+            f.force1x.highlight_time_range(item, annotation="calibration 1")
+
+            # Draws a line with annotation at the time the calibration was applied.
+            f.force1x.highlight_time_range(item.applied_at, annotation="calibration 1 applied")
+        """
+        import matplotlib.pyplot as plt
+
+        def to_seconds(timestamp):
+            start_ts = self.start if start is None else start
+
+            if isinstance(timestamp, str):
+                timestamp = to_timestamp(timestamp, start_ts, self.stop)
+
+            return (timestamp - start_ts) / 1e9
+
+        def get_time_range(time_item):
+            try:
+                return to_seconds(time_item.start), to_seconds(time_item.stop)  # Marker-like item?
+            except AttributeError:
+                pass
+
+            # Have to process single items separately, since otherwise strings might end up getting
+            # indexed in the last case. Note that single entry numpy arrays return false
+            # for np.isscalar, hence the check with ndim instead.
+            if np.ndim(time_item) == 0:
+                try:
+                    return to_seconds(time_item), None
+                except TypeError:
+                    raise TypeError(
+                        "Provided item must be either timestamp, two timestamps or an item with a "
+                        f"start and stop property, got {time_item} instead."
+                    )
+
+            try:
+                return to_seconds(time_item[0]), to_seconds(time_item[1])
+            except (TypeError, IndexError):
+                raise TypeError(
+                    "Provided item must be either timestamp, two timestamps or an item with a "
+                    f"start and stop property, got {time_item} instead."
+                )
+
+        start_time, stop_time = get_time_range(item)
+        if annotation:
+            self._annotate(start_time, annotation, annotation_direction, stop_time, color)
+
+        plt.axvline(start_time, label=label if not stop_time else "_", color=color, **kwargs)
+        if stop_time:
+            plt.axvline(stop_time, **kwargs)
+            plt.axvspan(start_time, stop_time, label="_", alpha=alpha, color=color, **kwargs)
```

Comment:
you're missing passing `color` in the 2nd `plt.axvline()`, if you pass in `color` for a time range it only gets applied to one of the lines
<img width="669" alt="image" src="https://github.com/user-attachments/assets/b8181c17-1b3f-408f-8505-c521df7093d0">

considering also the error I described above, I think it's better to not have `color` or `alpha` as explicit arguments here but just accept `**kwargs` and let the color cycler deal with it for the default case. We can still give a default `alpha` using:
```suggestion
        plt.axvline(start_time, label=label if not stop_time else "_", **kwargs)
        if stop_time:
            plt.axvline(stop_time, **kwargs)
            plt.axvspan(start_time, stop_time, label="_", **{"alpha": 0.5} | kwargs)
```

Reply:
Good catch!

I think I was initially passing it over `**kwargs`, but felt that explicitly adding `color` as an input argument makes the api more discoverable (that's probably when the bug snuck in). I've swapped it out now, but it seems that `c` is not a valid shorthand for `color` for `axvspan`.

The color cycler does not apply to `axvline` or `axhline`. I did look into incrementing the color cycle manually while I was working on this. It's possible to create a custom color cycler from `plt.rcParams['axes.prop_cycle']`, but it would require us maintaining the color cycle state ourselves, which might have unexpected corner cases in how it plays with the rest of the `mpl` ecosystem.